library/dash/src/main/java/com/google/android/exoplayer2/source/dash/manifest/DashManifestParser.java

```diff
@@ -1266,6 +1262,32 @@ protected static String parseEac3SupplementalProperties(List<Descriptor> supplem
     return MimeTypes.AUDIO_E_AC3;
   }
 
+  protected static String parseRole(List<Descriptor> roleDescriptors) {
```

Comment:
Making these non-static to allow overriding seems fine to me. Maybe call them `parseRoleDescriptors` and `parseAccessibilityDescriptors`? And move them up to be grouped with the other protected non-static methods.

Note that after the suggestions I made on the corresponding issue, these methods will return `@Role` and `@Accessibility` integers respectively.

Reply:
I'd be tempted to make a `parseSelectionFlags` method too, that can parse selection flags from the role descriptors. The default implementation will contain the little piece of code that decides whether to set `C.SELECTION_FLAG_DEFAULT` that's currently further up in this class.